src/manifolds/Stiefel.jl

```diff
@@ -55,12 +55,21 @@ A retraction based on the Caley transform, which is realized by using the
 """
 const CaleyRetraction = PadeRetraction{1}
 
+# This could maybe also be moved to Base.
 """
-    CaleyVectorTransport <: AbstractVectorTransportMethod
+    DifferentiatedRetraction{R<:AbstractRetractionMethod} <: AbstractVectorTransportMethod
 
-A vector transport that one obtains by differentiating a [`CaleyRetraction`](@ref).
+A type to specify a vector transport that is given by differentiating a retraction
```

Comment:
Could you add a generic formula here in docs? It looks like it would work for any manifold. Additionally, isn't `ParallelTransport` a `DifferentiatedRetraction` for the exponential retraction?

Also, maybe a different name would be better, `DifferentiatedRetraction` doesn't clearly indicate that it's a vector transport and is a bit vague.

Reply:
Yes, this would work for any manifold and yes, as far as I see, the DifferentiatedRetraction of exp is PT; just for fun we could also add this.
Currently it's placed within `Stiefel` since I did not find a good place to put it within `Manifolds.jl` since it is generic enough to even go over to `ManifoldsBase.jl` I think.

I changed the name to `DifferentiatedRetractionVectorTransport` – I initially felt that to be too long, but you're right that otherwise one does not see what it is. I also added the formula from Absil Mahony Sepulchre.

Reply:
Nice, it looks much more clear now. I think we can definitely put `DifferentiatedRetractionVectorTransport` in `ManifoldsBase.jl` and define `ParallelTransport` as an alias for differentiated retraction of the exponential map.

Reply:
Ok, I will do that.